src/mqt/bench/evaluation/evaluation_visualization.ipynb

```diff
@@ -0,0 +1,300 @@
+    "        if \"mapped_ibm_montreal\" in str_file:\n",
+    "            mapped_ibm_montreal_count += 1\n",
+    "        if \"mapped_ibm_washington\" in str_file:\n",
+    "            mapped_ibm_washington_count += 1\n",
+    "        if \"mapped_oqc_lucy\" in str_file:\n",
+    "            mapped_oqc_lucy_count += 1\n",
+    "        if \"mapped_rigetti\" in str_file:\n",
+    "            mapped_rigetti_count += 1\n",
+    "        if \"mapped_ionq\" in str_file:\n",
+    "            mapped_ionq_count += 1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9c53ad8e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perc_qiskit_indep = indep_qiskit / overall_count_indep\n",
+    "print(perc_qiskit_indep)\n",
+    "perc_tket_indep = indep_tket / overall_count_indep\n",
+    "print(perc_tket_indep)\n",
+    "print(perc_qiskit_indep + perc_tket_indep, overall_count_indep)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0abc68e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perc_nat_ibm = nat_ibm / overall_count_nat\n",
+    "print(perc_nat_ibm)\n",
+    "perc_nat_rigetti = nat_rigetti / overall_count_nat\n",
+    "print(perc_nat_rigetti)\n",
+    "perc_nat_oqc = nat_oqc / overall_count_nat\n",
+    "print(perc_nat_oqc)\n",
+    "perc_nat_ionq = nat_ionq / overall_count_nat\n",
+    "print(perc_nat_ionq)\n",
+    "print(sum([perc_nat_ibm, perc_nat_rigetti, perc_nat_oqc, perc_nat_ionq]), overall_count_nat)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bfc2767",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perc_montreal = mapped_ibm_montreal_count / overall_count_mapped\n",
+    "print(perc_montreal)\n",
+    "perc_washington = mapped_ibm_washington_count / overall_count_mapped\n",
+    "print(perc_washington)\n",
+    "perc_oqc = mapped_oqc_lucy_count / overall_count_mapped\n",
+    "print(perc_oqc)\n",
+    "perc_rigetti = mapped_rigetti_count / overall_count_mapped\n",
+    "print(perc_rigetti)\n",
+    "perc_ionq = mapped_ionq_count / overall_count_mapped\n",
+    "print(perc_ionq)\n",
+    "print(sum([perc_montreal, perc_washington, perc_oqc, perc_rigetti, perc_ionq]), overall_count_mapped)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "089acc25",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = [perc_washington, perc_rigetti, perc_montreal, perc_ionq, perc_oqc]\n",
+    "colors = [\"#B7E6A5\", \"#7CCBA2\", \"#46AEA0\", \"#089099\", \"#00718B\"]\n",
+    "labels = [\"IBMQ Washington (127)\", \"Rigetti Aspen M2 (80)\", \"IBMQ Montreal (27)\", \"IonQ Harmony (11)\", \"OQC Lucy (8)\"]\n",
+    "pies = plt.pie(data, labels=labels, autopct=\"%1.1f%%\", colors=colors)\n",
+    "for pie in pies[0]:\n",
+    "    pie.set_alpha(0.6)\n",
+    "plt.savefig(\"results/pie.pdf\", format=\"pdf\", bbox_inches=\"tight\")\n",
+    "plt.show()"
+   ]
```

Comment:
I just put this comment here because it is the first place where this suggestion fits, but what's coming next applies way more broadly to you repositories:

I am not a big fan of storing many pre-generated PDFs as part of our repositories as they quickly amount to a lot of files and just "lie around" in folders without much context.
Even more so, I am not a huge fan of Jupyter notebooks as they are really hard to read when not rendered properly by something like GitHub or your local IDE.

So the first thing I would advocate for, is moving as many utility functions as possible from the Jupyter notebook to a simple Python module. In this case here, that could be the `evaluation` module. Or it could be called `visualization` or something similar.
It would also be great if these visualisation methods could be somewhat covered with unit tests. That would probably require some kind of mocking/monkeypatching, but should be possible.
IMHO, the Jupyter notebooks should merely consist of some calls to library functions plus some added text for documentation purposes.

Now, I am a big fan of how we use Jupyter Notebooks in projects like QCEC and QMAP, where they are part of the documentation. See, e.g., https://github.com/cda-tum/qcec/blob/main/docs/source/Quickstart.ipynb, which gets rendered as https://qcec.readthedocs.io/en/latest/Quickstart.html in the documentation.
What happens here, is that the RtD documentation builder just executes the (clean) Jupyter notebooks and builds a documentation page out of that; including all the output.

All the evaluations we perform here (but also in the Predictor or the Problem Solver) would very nicely fit into their own documentation pages (assuming the generation of the respective images does not take hours).
This would eliminate the need for storing the individual PDFs in the repository as they would just be generated on-the-fly. In addition, documentation pages provide quite a lot more context than the individual PDF files lying around.

I guess what I am trying to say is: We should really get some RtD up and running cause it could significantly improve the way we can visualize data and results.

Reply:
I agree with you. The documentation improvement and shift to Read-the-Docs is on my todo-list for the next weeks.